pkg/controllers/state/cluster.go

```diff
@@ -221,11 +221,10 @@ func (c *Cluster) UpdateNodeClaim(nodeClaim *v1beta1.NodeClaim) {
 	c.mu.Lock()
```

Comment:
We're missing the update to the `Synced()` call here which validates that we don't have any `nodeClaimNametoProviderID` entries that have empty values